main/modes/mode_gamepad.c

```diff
@@ -71,43 +103,33 @@ void gamepadTouchCb(touch_event_t* evt);
 void gamepadAccelCb(accel_t* accel);
 void gamepadReportStateToHost(void);
 
-void setGamepadMainMenu(void);
+void setGamepadMainMenu(bool resetPos);
 void gamepadMainMenuCb(const char* opt);
 void gamepadMenuLoop(int64_t elapsedUs);
 void gamepadMenuButtonCb(buttonEvt_t* evt);
 void gamepadMenuTouchCb(touch_event_t* evt);
 void gamepadMenuAccelCb(accel_t* accel);
 void gamepadStart(display_t* disp, gamepadType_t type);
 
+static bool saveGamepadToggleSettings(gamepadToggleSettings_t* toggleSettings);
+static bool loadGamepadToggleSettings(gamepadToggleSettings_t* toggleSettings);
+
 //==============================================================================
 // Variables
 //==============================================================================
 
-
 static const char str_gamepadTitle[] = "Gamepad Type";
 static const char str_pc[] = "PC";
 static const char str_ns[] = "Switch";
+static const char str_touch_analog_on[] = "Touch: Digi+Analog";
+static const char str_touch_analog_off[] = "Touch: Digital Only";
+static const char str_accel_on[] = "Accel: On";
+static const char str_accel_off[] = "Accel: Off";
 static const char str_exit[] = "Exit";
+static const char KEY_GAMEPAD_TOGGLES[] = "gpts";
 
-typedef struct
-{
-    font_t mmFont;
-    meleeMenu_t* menu;
-    display_t* disp;
-    gamepadScreen_t screen;
-    uint8_t selectedGamepadType;
-} gamepadMenu_t;
-
-typedef struct
-{
-    display_t* disp;
-    hid_gamepad_report_t gpState;
-    hid_gamepad_ns_report_t gpNsState;
-    font_t ibmFont;
-    uint8_t gamepadType;
-    bool isPluggedIn;
-} gamepad_t;
-
+gamepadMenu_t* gm;
```

Comment:
`gm` and `gamepadToggleSettings` should be members of `gamepad_t`. It saves a tiny bit of RAM and is generally cleaner

Reply:
And they don't have to be pointers as members of `gamepad_t` either! They'll all get alloc'd and free'd together

Reply:
I can wrap `gm` into `gamepad_t` but I wanted to have `gamepadToggleSettings` separate so I could save and load it as an NVS blob

Reply:
Oh wait I misunderstood you and only realized partway through fixing this. The way I ended up doing it puts all of `gm`'s stuff into `gamepad`, removing `gamepadMenu_t`, and makes `gamepadToggleSettings` a pointer inside `gamepad`